lua/wire/server/wirelib.lua

```diff
@@ -363,39 +339,31 @@ function WireLib.AdjustSpecialOutputs(ent, names, types, descs)
 	end
 
 
-	local i = 0
-	for n,v in ipairs(names) do
+	for n, v in ipairs(names) do
 		local name, desc, tp = ParsePortName(v, types[n] or "NORMAL", descs and descs[n])
 
-		if (ent_ports[name]) then
-			if tp ~= ent_ports[name].Type then
+		local port = ent_ports[name]
+		if port then
+			if tp ~= port.Type then
 				WireLib.DisconnectOutput(ent, name)
-				ent_ports[name].Type = tp
+				port.Type = tp
 			end
-			WireLib.RemoveOutPort(ent, name)
-			ent_ports[name].Keep = true
-			ent_ports[name].Desc = desc
+			port.Keep = true
+			port.Num = n
+			port.Desc = desc
 		else
-			i = i + 1
```

Comment:
Only issue i see with this pr is the removal of the `i = i + 1` logic, before it only incremented when the port didn't exist, now n is directly bound to ipairs which seems like different behavior than before

Removing idx seems fine

Reply:
Oh i see, in the original PR the `i = i + 1` logic was introduced, did this cause the bug this pr is aiming to fix?

Reply:
Yeah, looks like the Num was getting screwed up when existing ports were reused after an AdjustPorts.

Reply:
AdjustSpecialInputs has the correct behavior. https://github.com/wiremod/wire/pull/3303/files#diff-11f733e8c74f850d57c2ed245e889bf04028edfa9f155d3c91bb44550d46c27cL299